Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -6245,3 +6245,205 @@ def update_with_app_data(self, app_data, mapper_name, mapping_type=None):
         if not isinstance(app_data, dict):
             app_data = safe_load_json(app_data)
         self._user_profile = demisto.mapObject(app_data, mapper_name, mapping_type)
+
+
+class IAMCommand:
+    def __init__(self, is_create_enabled, is_disable_enabled, is_update_enabled,
+                 create_if_not_exists, mapper_in, mapper_out):
+        """ The IAMCommand c'tor
+
+        :param is_create_enabled: (bool) Whether or not the `iam-create-user` command is enabled in the instance
+        :param is_disable_enabled: (bool) Whether or not the `iam-disable-user` command is enabled in the instance
+        :param is_update_enabled: (bool) Whether or not the `iam-update-user` command is enabled in the instance
+        :param create_if_not_exists: (bool) Whether or not to create a user if does not exist in the application
+        :param mapper_in: (str) Incoming mapper from the application to Cortex XSOAR
+        :param mapper_out: (str) Outgoing mapper from the Cortex XSOAR to the application
+        """
+        self.is_create_enabled = is_create_enabled
+        self.is_disable_enabled = is_disable_enabled
+        self.is_update_enabled = is_update_enabled
+        self.create_if_not_exists = create_if_not_exists
+        self.mapper_in = mapper_in
+        self.mapper_out = mapper_out
+
+    def get_mapping_fields(self, client):
+        """ Creates and returns a GetMappingFieldsResponse object of the user schema in the application
+
+        :param client: (Client) The integration Client object that implements a get_app_fields() method
+        :return: (GetMappingFieldsResponse) An object that represents the user schema
+        """
+        app_fields = client.get_app_fields()
+        incident_type_scheme = SchemeTypeMapping(type_name=IAMUserProfile.INDICATOR_TYPE)
+
+        for field, description in app_fields.items():
+            incident_type_scheme.add_field(field, description)
+
+        return GetMappingFieldsResponse([incident_type_scheme])
```

Comment:
This whole functionality can move to the client.

Reply:
why? it's also supposed to be the same in every integration (and can be overwritten in case not)

Reply:
you right, they are already doing something similar in the client.
The reason I say it, is because this way the client will have well defined signature.
The client will have to return GetMappingFieldsResponse with fields otherwise you will fail the function